### ISSUE TITLE
default_collate问题

### ISSUE THREAD

**Issue by lcylcy** [Oneflow-Inc/oneflow#6317]
from flow.utils.data.dataloader import default_collate

a = [{"text":np.random.rand(2,4),
      "label":0,
      "position": np.random.rand(2,0)},
      {"text":np.random.rand(2,4),
      "label":0,
      "position": np.random.rand(2,0)}]
b = default_collate(a)
print(b)

发生Floating point exception (core dumped)

**Comment by Flowingsun007** [comment]
这个问题经过定位，发现不是default_collate的问题，主要是default_collate里用到了stack，而stack对输入为0-shape tensor构成的list时会bug：`Floating point exception`。

以下是最小复现代码：
```python
>>> import oneflow as flow
>>> import numpy as np
>>> a = np.random.rand(2,0)
>>> x1 = flow.tensor(a)
>>> x2 = flow.tensor(a)
>>> batch = [x1, x2]
>>> y = flow.stack(batch, dim=0)
>>> Floating point exception
```

同样操作，pytorch可以正常运行：
```python
>>> a
array([], shape=(2, 0), dtype=float64)
>>> x1 = torch.tensor(a)
>>> x2 = torch.tensor(a)
>>> batch = [x1, x2]
>>> y = torch.stack(batch, dim=0)
>>> y
tensor([], size=(2, 2, 0), dtype=torch.float64)
```

**Comment by wyg1997** [comment]
原因是 concat 支持 0shape 计算时检察不够细致，在Compute 中推导时出现了除 0 的错误。

解决办法是在 https://github.com/Oneflow-Inc/oneflow/blob/master/oneflow/user/kernels/concat_kernel.cpp#L53 加一个对 
 `output.shape().elemt_cnt()` 的特判就行了。
